packages/std/src/math/decimal.rs

```diff
@@ -55,6 +56,12 @@ impl Decimal {
     pub fn is_zero(&self) -> bool {
         self.0 == 0
     }
+
+    /// Returns the approximate square root as a Decimal.
+    pub fn sqrt(self) -> Self {
+        // TODO: Should we have better precision here?
+        Decimal(self.0.isqrt() * 1_000_000_000)
```

Comment:
Good point. `sqrt(a) * 10^9` is also `sqrt(a * 10^9^2)` = `sqrt(a * 10^18)`. This would reduce rounding but risk overflows.

Reply:
What we can do: We need to multiply 9 times by 10 outside of the sqrt or 9 times by 100 inside the sqrt. Now we can do checked multiplications inside as long as they work. Once we hit an overflow we stop and continue outside.